arch/arm/src/common/arm_fpucmp.c

```diff
@@ -25,70 +25,35 @@
 #include <nuttx/config.h>
 
 #include <stdint.h>
-#include <stdbool.h>
 #include <string.h>
-#include <debug.h>
-
 #include <nuttx/irq.h>
-#include <arch/board/board.h>
 
 #include "arm_internal.h"
-#include "bambino-200e.h"
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* Configuration ************************************************************/
-
-#undef HAVE_FPU
-#if defined(CONFIG_ARCH_FPU) && defined(CONFIG_TESTING_OSTEST_FPUSIZE) && \
-    defined(CONFIG_SCHED_WAITPID)
-#    define HAVE_FPU 1
-#endif
-
-#ifdef HAVE_FPU
-
-#if CONFIG_TESTING_OSTEST_FPUSIZE != (4*SW_FPU_REGS)
-#  error "CONFIG_TESTING_OSTEST_FPUSIZE has the wrong size"
-#endif
+#ifdef CONFIG_ARCH_FPU
 
 /****************************************************************************
- * Private Data
+ * Public Functions
  ****************************************************************************/
 
-static uint32_t g_saveregs[XCPTCONTEXT_REGS];
-
 /****************************************************************************
- * Public Functions
+ * Name: up_fpucmp
+ *
+ * Description:
+ *   compare FPU areas from thread context
+ *
  ****************************************************************************/
 
-/* Given an array of size CONFIG_TESTING_OSTEST_FPUSIZE, this function will
- * return the current FPU registers.
- */
-
-void arch_getfpu(FAR uint32_t *fpusave)
+bool up_fpucmp(FAR const void *saveregs1, FAR const void *saveregs2)
 {
-  irqstate_t flags;
-
-  /* Take a snapshot of the thread context right now */
-
-  flags = enter_critical_section();
-  arm_saveusercontext(g_saveregs);
+  FAR const uint32_t *regs1 = saveregs1;
+  FAR const uint32_t *regs2 = saveregs2;
```

Comment:
let's remove `FAR` as Mr. @xiaoxiang781216 is cleaning things up for ARM

Reply:
Done